nvflare/private/fed/client/client_runner.py

```diff
@@ -278,8 +300,17 @@ def _process_task(self, task: TaskAssignment, fl_ctx: FLContext) -> Shareable:
                 fl_ctx=fl_ctx,
                 msg=f"submit result: {ReturnCode.EXECUTION_RESULT_ERROR}",
             )
+        except UnsafeJobError:
+            self.log_exception(fl_ctx, f"UnsafeJobError from executor {executor_name}")
+            executor.unsafe = True
```

Comment:
Who / which function is raising the UnsafeJobError? The "unsafe" flag should be set whenever this UnsafeJobError exception is raised, instead of setting it in the exception catch.

Reply:
Agreed. See my comments in response to Chester's.

Reply:
Anyone can detect a job is unsafe. The exception UnsafeJobError is mainly used by filters and executor, since this is how our customers are already doing (raise exception to end job).